### PR TITLE
refactor: remove dead code

### LINKED ISSUE #1825: Refactor: remove un-used exports
- [x] Remove unused exports
- [x] Create CI job once removed that ensures none are allowed

<details>
<summary>All unused modules</summary>

```
72 modules with unused exports
src/routes.tsx: Route
src/background/vault-types.ts: VaultMessage, GetWallet, MakeWallet, CreateNewAccount, SignOut, LockWallet
src/background/vault.ts: getVault, vaultReducer
src/common/asset-types.ts: MetaDataMethodNames, FungibleTokenOptions, MetaDataNames
src/common/clarity-prism.ts: clarity
src/common/constants.ts: STACKS_MARKETS_URL, SIP_010, DEFAULT_TESTNET_SERVER, DEFAULT_REGTEST_SERVER, DEFAULT_MAINNET_SERVER
src/common/message-types.ts: AuthenticationRequestMessage, TransactionRequestMessage
src/common/onboarding-data.ts: faqs
src/common/stacks-utils.ts: makeAssetInfo, getTicker
src/common/storage.ts: getRequestInfo
src/common/token-utils.ts: SIP010TransferResponse
src/common/types.ts: ONBOARDING_PROGRESS, CHANGE_PAGE, SAVE_KEY, SAVE_AUTH_REQUEST, SET_MAGIC_RECOVERY_CODE, SET_USERNAME, SET_ONBOARDING_PATH, persistedScreens, OnboardingState, OnboardingActions
src/common/utils.ts: getRandomWord, stringToHslColor, isUtf8, isHex
src/common/dev/types/index.ts: AppManifest
src/common/hooks/use-color-mode.ts: THEME_STORAGE_KEY
src/common/hooks/account/use-account-info.ts: useFetchAccountData
src/common/transactions/post-condition-utils.ts: getTitleFromConditionCode
src/common/validation/currency-schema.ts: curencyAmountSchema
src/common/validation/validate-password.ts: ValidatedPassword
src/components/color-modes.tsx: colorGet, color, LightMode, DarkMode, Base, ColorModes, ColorModeContext, ColorModeProvider
src/components/extension-button.tsx: ExtensionButton
src/components/item-hover.tsx: ItemHover
src/components/link.tsx: LinkProps
src/components/list-item.tsx: ListItem
src/components/stx-avatar.tsx: StxAvatar
src/components/typography.tsx: titleStyles, Pretitle
src/components/accounts/account-avatar.tsx: AccountAvatar
src/components/vector/index.tsx: AppsIcon, BlockchainIcon, EncryptionIcon, PrivateIcon, EyeIcon
src/features/account-avatar/account-avatar.tsx: AccountAvatarWithNameInner
src/features/devtool/devtools.tsx: Devtools
src/features/errors/error-boundary.tsx: withErrorBoundary, ErrorBoundaryPropsWithComponent, ErrorBoundaryPropsWithRender, ErrorBoundaryPropsWithFallback, ErrorBoundaryProps
src/pages/choose-account/components/accounts.tsx: AccountItem
src/pages/save-your-secret-key/save-your-key-view.tsx: SecretKeyMessage, SecretKeyCard, SecretKeyActions
src/pages/send-tokens/components/fee-fetcher.tsx: FeeFetcher
src/pages/send-tokens/components/asset-search/asset-search-result.tsx: AssetResult
src/pages/send-tokens/hooks/use-send-form.ts: useSendForm
src/pages/transaction-signing/components/actions.tsx: MinimalErrorMessage
src/pages/transaction-signing/components/contract-call-details.tsx: ContractCallDetailsSuspense
src/pages/transaction-signing/components/error.tsx: ErrorMessageProps
src/pages/transaction-signing/components/popup-header.tsx: PopupHeaderFallback, PopupHeaderSuspense
src/pages/transaction-signing/components/transaction-error.tsx: TransactionErrorSuspense
src/pages/transaction-signing/components/post-conditions/fungible-post-condition.tsx: FungiblePostConditionComponentSuspense
src/pages/transaction-signing/components/post-conditions/post-condition.tsx: PostConditionComponentSuspense
src/pages/transaction-signing/components/post-conditions/post-conditions-list.tsx: PostConditionsSuspense
src/store/query.ts: queryAtom
src/store/accounts/account-activity.hooks.ts: useCurrentAccountTxids, useCurrentAccountLocalTxs
src/store/accounts/account-activity.ts: currentAccountLocallySubmittedTxsRootState
src/store/accounts/account-activity.utils.ts: getAddressFromPublicKeyHash
src/store/accounts/api.ts: PaginatedResults, accountBalancesAnchoredClient
src/store/accounts/nonce.ts: currentAccountLocalNonceState
src/store/assets/asset-search.ts: searchResultState
src/store/assets/asset.hooks.ts: useFungibleTokenState, useNonFungibleTokenState
src/store/assets/tokens.ts: baseAssetsAnchoredState, baseAssetsUnanchoredState, assetsUnanchoredState
src/store/assets/utils.ts: getLocalData, setLocalData
src/store/common/api-clients.ts: apiClients
src/store/common/common.hooks.ts: WritableAtom, useDynamicUseAtom_UNSAFE, useErrorHandler
src/store/common/constants.ts: DEFAULT_POLLING_INTERVAL, DEFAULT_FEE_RATE_MULTIPLIER
src/store/network/networks.ts: latestBlockHeightState, networkInfoState
src/store/onboarding/onboarding.ts: OnboardingState
src/store/transactions/index.ts: signedStacksTransactionState, isUnauthorizedTransactionState
src/store/transactions/local-transactions.ts: tokenTransferTransaction, ftTokenTransferTransactionState, localTransactionIsStxTransferState
src/store/transactions/raw.hooks.ts: useRawTxState
src/store/transactions/requests.hooks.ts: useTransactionRequestCustomFeeRate
src/store/transactions/transaction.hooks.ts: useTransactionContractSource, useMakeAssetTransfer, useLocalTransactionState
src/store/transactions/utils.ts: getTxByteLength
src/store/ui/ui.hooks.ts: showNetworksStore, showAccountsStore, showTxSettingsStore
tests/global-setup.ts: default
tests/global-teardown.ts: default
tests/integration/utils.ts: debug
tests/mocks/heystack.ts: GetRequests, getRequests, postRequests
tests/mocks/index.ts: STX_TRANSFER_DECODED
tests/mocks/heystack/data.ts: HEY_CONTRACT_EVENTS
```

</details>

### PR DESCRIPTION
**Pull request by kyranjamie**
> Try out this version of the Hiro Wallet - download [extension builds](https://github.com/blockstack/ux/actions/runs/1358161675).

Perhaps my best pull request ever? 🌈  

So much dead code. Can't even believe that this will work.

### CODE REVIEW

**Review by fbwoolf — APPROVED** [review]

File: src/store/assets/utils.ts [hunk 1]
```diff
@@ -3,35 +3,6 @@ import BigNumber from 'bignumber.js';
 import type { Asset } from '@common/asset-types';
 import type { AccountBalanceResponseBigNumber } from '@store/accounts/types';
 
-function makeKey(networkUrl: string, address: string, name: string, key: string): string {
-  return `${networkUrl}__${address}__${name}__${key}`;
-}
-
-export function getLocalData(options: {
```
Comment (beguene) [inline src/store/assets/utils.ts:10]: getLocalData is used in assetMetaDataState
https://github.com/blockstack/stacks-wallet-web/blob/main/src/store/assets/fungible-tokens.ts#L40 which is itself used.
Have you tested the app with your changes ?

**Review by beguene — DISMISSED** [review]

**Comment by beguene** [comment]
@kyranjamie Everything else looks good 👍

File: src/store/assets/utils.ts (hunk 1 above)
Comment (kyranjamie) [reply src/store/assets/utils.ts:10]: I have not yet, no. This is very interesting, because it compiles fine 🤔

Comment (kyranjamie) [reply src/store/assets/utils.ts:10]: Changed back to draft, as I haven't _really_ tested. But in this instance, it turns out there are two`getLocalData`s. A very keen eye'd spot, though.

Comment (beguene) [reply src/store/assets/utils.ts:10]: It looks like it does not recognise atom dependencies. I checked all the other deletions you made, they are all unused (which is perfect).
So you can add this 'getLocalData ' again, and we should be a in a good state.

Comment (kyranjamie) [reply src/store/assets/utils.ts:10]: It does. The `getLocalData` used in `fungible-tokens.ts` is here.

https://github.com/blockstack/stacks-wallet-web/blob/cb0b5d872b781b9f5626cb731cb501653e470751/src/common/store-utils.ts#L25-L30

The compiler would error if it were imported but undefined.

**Comment by kyranjamie** [comment]
Looks like this has broken test-app

**Pull request merged**